lib/actions/purgeTable.js

```diff
@@ -0,0 +1,77 @@
+let tableName = null;
+let dynamodb = null;
+
+const findPrimaryKey = () => {
+  const params = {
+    TableName: this.tableName
+  }
+
+  return new Promise((resolve, reject) => {
+    this.dynamodb.describeTable(params).promise()
+      .then(tableDescription => {
+        resolve({
+          primaryKey: tableDescription.Table.KeySchema.filter((element) => element.KeyType === "HASH")[0].AttributeName
+        })
+      })
+  })
+}
+
+const deleteAllElements = (deleteRequest) => {
+  const deleteRequests = [];
+
+  for (item of deleteRequest.items) {
+    let params = {
+      TableName: this.tableName,
+      Key: {}
+    }
+
+    params.Key[deleteRequest.primaryKey] = item[deleteRequest.primaryKey];
```

Comment:
In the `for` loop, `item` is not declared (missing const).

But I would suggest shorter code for this:

```
const primaryKey = deleteRequest.primaryKey

return Promise.all(deleteRequest.items.map(item => this.dynamodb.deleteItem({
  TableName: this.tableName,
  Key: {
    [primaryKey]: item.primaryKey,
  },
}).promise())
```